lib/ansible/plugins/filter/core.py

```diff
@@ -532,6 +532,16 @@ def dict_to_list_of_dict_key_value_elements(mydict):
     return ret
 
 
+def list_of_dict_key_value_elements_to_dict(mylist, key_name='key', value_name='value'):
+    ''' takes a list of dicts with each having a 'key' and 'value' keys, and transforms the list into a dictionary,
+        effectively as the reverse of dict2items '''
+
+    if not isinstance(mylist, list):
```

Comment:
some filters output generators or sets, they should be usable here, not sure if they match list
`if isinstance( ,string_types) and not isinstance(, Sequence)` should cover all 'usable'